mdspan_copy/mdspan_copy.md

```diff
@@ -0,0 +1,119 @@
+
+This requirement is likely undesirable for a generalized copy algorithm.
+
+There is a similar argument against simply generalizing `std::linalg::copy`. In addition to the freedom of `std::linalg::copy` to arbitrarily value or direct-initializing values, using the linear algebra version of copy would require the use of unnecessary includes and namespaces. It seems not very ergonomic for a user to have to use `std::linalg::copy` and include `<linalg>` even if the `mdspan` operations they are performing are unrelated to linear algebra.
+
+## What the proposal does not include
+
+There are a few additions that are analogous to existing standard algorithms that are not included in this proposal, both to keep the proposal small and because some of these algorithms do not make sense in the context of `mdspan`s:
+
+* `std::move`: Perhaps this should be included for completeness's sake. However, it doesn't seem applicable to the typical usage of `mdspan`.
+* `(copy|fill)_n`: As a multidimensional view `mdspan` does not in general follow a specific ordering. Memory ordering may not be obvious to calling code, so it's not even clear how these would work. Any applications intending to copy a subset of `mdspan` should use call `copy` on the result of `submdspan`.
+* `copy_backward`: As above, there is no specific ordering. A similar effect could be achieved via transformations with a custom layout, similar to `layout_transpose` in [@P1673R13].
+* Other algorithms, include `std::for_each`. `for_each` in particular is a desirable but brings in many unanswered questions that should be addressed in a different paper.
+
+# Wording
+
+```c++
+template<class SrcElementType, class SrcExtents, class SrcLayoutPolicy, class SrcAccessorPolicy,
+         class DstElementType, class DstExtents, class DstLayoutPolicy, class DstAccessorPolicy>
+void copy(mdspan<SrcElementType, SrcExtents, SrcLayoutPolicy, SrcAccessorPolicy> src,
+          mdspan<DstElementType, DstExtents, DstLayoutPolicy, DstAccessorPolicy> dst);
+
+template<class ExecutionPolicy, class SrcElementType, class SrcExtents, class SrcLayoutPolicy, class SrcAccessorPolicy,
+         class DstElementType, class DstExtents, class DstLayoutPolicy, class DstAccessorPolicy>
+void copy(ExecutionPolicy&& policy, mdspan<SrcElementType, SrcExtents, SrcLayoutPolicy, SrcAccessorPolicy> src,
+          mdspan<DstElementType, DstExtents, DstLayoutPolicy, DstAccessorPolicy> dst);
+```
+
+[1]{.pnum} *Constraints:*
+
+  * [1.1]{.pnum} `std::is_assignable_v<typename mdspan<SrcElementType, SrcExtents, SrcLayoutPolicy, SrcAccessorPolicy>::reference, typename mdspan<DstElementType, DstExtents, DstLayoutPolicy, DstAccessorPolicy>::reference>` is `true`.
+
+  * [1.2]{.pnum} `mdspan<SrcElementType, SrcExtents, SrcLayoutPolicy, SrcAccessorPolicy>::rank()` equals `mdspan<DstElementType, DstExtents, DstLayoutPolicy, DstAccessorPolicy>::rank()`.
+
```

Comment:
I think we either need a constraint or a Mandate on extents compatibility
```suggestion
[2]{.pnum} *Mandates:* `is_constructible_v<DstExtents, SrcExtents>` 
```

This will check for rank but also for matching static extents if necessary. I.e. if you have `mdspan<int, extents<int,2>` and `mdspan<int, extents<int,4>` it will trigger.

Reply:
Ah, I was looking for how to word that (wrt able to check static extents). For now, I think I will add it as an extent, because maybe you would want to SFINAE on it? Also it can replace 1.2

Reply:
I added it as a constraint